### PR TITLE
trie: fix error in node decoding

### PR DESCRIPTION
**Pull request by holiman**
Fixes an error in trie decoding. 
The flaw was found by Lukasz Matczak

### CODE REVIEW

File: trie/trie_test.go [hunk 1]
```diff
@@ -614,3 +614,16 @@ func updateString(trie *Trie, k, v string) {
 func deleteString(trie *Trie, k string) {
 	trie.Delete([]byte(k))
 }
+func TestDecodeNode(t *testing.T){
```
Comment (fjl) [inline trie/trie_test.go]: Please add newline here

File: trie/trie_test.go [hunk 2]
```diff
@@ -614,3 +614,16 @@ func updateString(trie *Trie, k, v string) {
...
+	var (
+		hash = make([]byte, 20)
+		elems = make([]byte, 20)
+	)
+	for i := 0; i < 100000000 ; i++{
+		rand.Read(hash)
+		rand.Read(elems)
+		decodeNode(hash, elems, 1)
+	}
+
```
Comment (fjl) [inline trie/trie_test.go]: Please remove newline here. You may also copy paste it above the function to resolve my other comment.

**Comment by holiman** [comment]
Addressed the concerns, also made the test run fewer times, to take 2s instead of 40s on my laptop

File: trie/trie_test.go (hunk 2 above)
Comment (holiman) [reply trie/trie_test.go]: Shit, I didn't consider reusing the linebreak. I just blindly created one and destroyed another - what a waste :(

**Pull request merged**